src/Grpc.AspNetCore.Server/Compression/GzipCompressionProvider.cs

```diff
@@ -0,0 +1,64 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System.IO;
+using System.IO.Compression;
+
+namespace Grpc.AspNetCore.Server.Compression
+{
+    /// <summary>
+    /// GZIP compression provider.
+    /// </summary>
+    public class GzipCompressionProvider : ICompressionProvider
+    {
+        private readonly CompressionLevel _compressionLevel;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GzipCompressionProvider"/> class with the specified <see cref="CompressionLevel"/>.
+        /// </summary>
+        /// <param name="compressionLevel">The compression level to use when compressing data.</param>
+        public GzipCompressionProvider(CompressionLevel compressionLevel)
```

Comment:
What about per call compression level settings? https://github.com/grpc/grpc/blob/master/doc/compression_cookbook.md#compression-level-in-call-responses

Reply:
Actually I think it's sufficient that we provide functionality through WriteFlags.NoCompress.